builtin/credential/approle/path_role.go

```diff
@@ -0,0 +1,1544 @@
+	// should never expire. The token should be renewed within the duration
+	// specified by this value. The renewal duration will be fixed if the
+	// value is not modified on the role. If the `Period` in the role is modified,
+	// a token will pick up the new value during its next renewal.
+	Period time.Duration `json:"period" mapstructure:"period" structs:"period"`
+}
+
+// rolePaths creates all the paths that are used to register and manage an role.
+//
+// Paths returned:
+// role/ - For listing all the registered roles
+// role/<role_name> - For registering an role
+// role/<role_name>/policies - For updating the param
+// role/<role_name>/secret-id-num-uses - For updating the param
+// role/<role_name>/secret-id-ttl - For updating the param
+// role/<role_name>/token-ttl - For updating the param
+// role/<role_name>/token-max-ttl - For updating the param
+// role/<role_name>/bound-secret-id - For updating the param
+// role/<role_name>/bound-cidr-list - For updating the param
+// role/<role_name>/period - For updating the param
+// role/<role_name>/role-id - For fetching the role_id of an role
+// role/<role_name>/secret-id - For issuing a secret_id against an role, also to list the secret_id_accessorss
+// role/<role_name>/secret-id/<secret_id_accessor> - For reading the properties of, or deleting a secret_id
+// role/<role_name>/custom-secret-id - For assigning a custom SecretID against an role
+func rolePaths(b *backend) []*framework.Path {
+	return []*framework.Path{
+		&framework.Path{
+			Pattern: "role/?",
+			Callbacks: map[logical.Operation]framework.OperationFunc{
+				logical.ListOperation: b.pathRoleList,
+			},
+			HelpSynopsis:    strings.TrimSpace(roleHelp["role-list"][0]),
+			HelpDescription: strings.TrimSpace(roleHelp["role-list"][1]),
+		},
+		&framework.Path{
+			Pattern: "role/" + framework.GenericNameRegex("role_name"),
+			Fields: map[string]*framework.FieldSchema{
+				"role_name": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Description: "Name of the role.",
+				},
+				"bound_secret_id": &framework.FieldSchema{
+					Type:        framework.TypeBool,
+					Default:     true,
+					Description: "Impose secret_id to be presented when logging in using this role. Defaults to 'true'.",
+				},
+				"bound_cidr_list": &framework.FieldSchema{
+					Type: framework.TypeString,
+					Description: `Comma separated list of CIDR blocks, if set, specifies blocks of IP
+addresses which can perform the login operation`,
+				},
+				"policies": &framework.FieldSchema{
+					Type:        framework.TypeString,
+					Default:     "default",
+					Description: "Comma separated list of policies on the role.",
+				},
+				"secret_id_num_uses": &framework.FieldSchema{
+					Type:        framework.TypeInt,
+					Description: "Number of times a SecretID can access the role, after which the SecretID will expire.",
+				},
+				"secret_id_ttl": &framework.FieldSchema{
```

Comment:
Same with the rest of these TTL params, might want to mention the default is zero which means fall back to mount/system defaults.